monsterCatalog.py

```diff
@@ -188,180 +190,143 @@ def monsters():
+
+
+def check_state_parameter(returned_state):
+    flask_state = flask.session['state']
+
+    if returned_state != flask_state:
+        bad_state_response = flask.jsonify('Invalid state parameter.')
+        bad_state_response.status_code = 401
+        return bad_state_response
+
+
+def load_fb_app_data():
     app_id = json.loads(open('fb_client_secrets.json', 'r').read())[
         'web']['app_id']
     app_secret = json.loads(
         open('fb_client_secrets.json', 'r').read())['web']['app_secret']
-    url = 'https://graph.facebook.com/oauth/access_token?grant_type=fb_exchange_token&client_id=%s&client_secret=%s&fb_exchange_token=%s' % (
-        app_id, app_secret, access_token)
-    h = httplib2.Http()
-    result = h.request(url, 'GET')[1]
-
-    # Use token to get user info from API
-    userinfo_url = "https://graph.facebook.com/v2.4/me"
-    # strip expire tag from access token
-    token = result.split("&")[0]
-
-    url = 'https://graph.facebook.com/v2.4/me?%s&fields=name,id,email' % token
-    h = httplib2.Http()
-    result = h.request(url, 'GET')[1]
-    # print "url sent for API access:%s"% url
-    # print "API JSON result: %s" % result
-    data = json.loads(result)
-    flask.session['provider'] = 'facebook'
-    flask.session['username'] = data["name"]
-    flask.session['email'] = data["email"]
-    flask.session['facebook_id'] = data["id"]
-
-    # The token must be stored in the login_session in order to properly logout, let's strip out the information before the equals sign in our token
-    stored_token = token.split("=")[1]
-    flask.session['access_token'] = stored_token
-
-    # Get user picture
-    url = 'https://graph.facebook.com/v2.4/me/picture?%s&redirect=0&height=200&width=200' % token
-    h = httplib2.Http()
-    result = h.request(url, 'GET')[1]
-    data = json.loads(result)
-
-    flask.session['picture'] = data["data"]["url"]
-
-    # see if user exists
-    user_id = getUserID(flask.session['email'])
+    return app_id, app_secret
+
+
+def exchange_fb_token(short_lived_access_token):
+    """Exchange short-lived access token for long-lived one."""
+    app_id, app_secret = load_fb_app_data()
+    fb_token_exchange_url = 'https://graph.facebook.com/oauth/access_token'
+    fb_token_exchange_params = {
+        'grant_type': 'fb_exchange_token',
+        'client_id': app_id,
+        'client_secret': app_secret,
+        'fb_exchange_token': short_lived_access_token
+    }
+    fb_token_exchange_result = requests.get(fb_token_exchange_url,
+                                            params=fb_token_exchange_params)
+    fb_token_exchange_text = fb_token_exchange_result.text
+    long_lived_access_token = fb_token_exchange_text.split("&")[0].split('=')[1]
+    return long_lived_access_token
+
+
+def get_fb_user_data(long_lived_access_token):
+    """Using long-lived access token, query Facebook Graph API for user data."""
+    fb_user_query_url = 'https://graph.facebook.com/v2.4/me'
```

Comment:
This can be a constant stored outside of the function.  Something like FB_USER_QUERY_URL at the top of the module.